### PR TITLE
ROC-7139 refine automatic stay claim

### PR DESCRIPTION
**Pull request by daveoh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7139

### Change description ###
Fix automatically staying cases incorrectly.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    STAY_CLAIM(CaseEvent.STAY_CLAIM,
+        AppInsightsEvent.CLAIM_STAYED,
+        (responseDate -> {
+            return QueryBuilders.boolQuery()
+                .must(QueryBuilders.termQuery("state", ClaimState.OPEN.getValue()))
+                .must(QueryBuilders.rangeQuery("data.respondents.value.responseSubmittedOn").lte(responseDate))
+                .mustNot(QueryBuilders.existsQuery("data.respondents.value.paidInFullDate"))
+                .mustNot(QueryBuilders.existsQuery("data.respondents.value.claimantResponse.submittedOn"))
+                .must(QueryBuilders.rangeQuery("data.submittedOn").gte(DateUtils.DATE_OF_5_0_0_RELEASE));
+        }),
```
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java]: (And for `WAITING_TRANSFER` if implemented)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        this.query = query;
+        this.triggerEvents = Collections.emptySet();
+        this.ignoredEvents = Collections.emptySet();
+    }
+
+    private CaseEvent caseEvent;
+    private AppInsightsEvent appInsightsEvent;
+    private Function<LocalDate, QueryBuilder> query;
+    private Set<CaseEvent> triggerEvents;
+    private Set<CaseEvent> ignoredEvents;
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java]: Should be above the constructor

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionServiceTest.java [hunk 3]
```diff
@@ -68,14 +75,36 @@
...
+
     @Before
     public void setUp() {
-        when(environment.getProperty(any(String.class))).thenReturn("10");
+        when(stateTransition.name()).thenReturn("STATE_TRANSITION");
+        when(stateTransition.getQuery()).thenReturn(localDate -> QueryBuilders.termQuery("date", localDate));
+        when(stateTransition.getCaseEvent()).thenReturn(caseEvent);
+        when(stateTransition.getAppInsightsEvent()).thenReturn(appInsightsEvent);
+
+        when(environment.getProperty(any(String.class))).thenReturn(deadline.toString());
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionServiceTest.java]: `anyString()`

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java [hunk 4]
```diff
@@ -64,7 +64,9 @@
     CREATE_CITIZEN_CLAIM("CreateCitizenClaim"),
     REVIEWED_PAPER_RESPONSE("ReviewedPaperResponse"),
     MEDIATION_FAILED("FailedMediation"),
-    MEDIATION_SUCCESSFUL("SuccessfulMediation");
+    MEDIATION_SUCCESSFUL("SuccessfulMediation"),
+    ATTACH_SCANNED_DOCS("attachScannedDocs"),
```
[the remark this answers is not in the document]
Comment (daveoh) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java]: I completely agree. This is an existing event that had not been defined in the claim store, just CCD. I would imagine that further data migration would be needed to change the id in CCD?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java [hunk 5]
```diff
@@ -75,11 +87,13 @@ public ScheduledStateTransitionService(
...
+        this.caseEventsApi = caseEventsApi;
+        this.authTokenGenerator = authTokenGenerator;
     }
 
     @PostConstruct
     public void init() {
-        for (StateTransition stateTransition : StateTransition.values()) {
+        for (StateTransition stateTransition : StateTransitions.values()) {
             getStateTransitionDaysProperty(stateTransition);
         }
```
[the remark this answers is not in the document]
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java]: I guess it depends on the coding standards as arguments can be made for both in terms of readability.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java [hunk 6]
```diff
@@ -110,6 +128,40 @@ public void transitionClaims(LocalDateTime runDateTime, User user, StateTransiti
...
+                    }
+
+                    return stateTransition.getTriggerEvents().contains(event);
+                }
+                return false;
+
+            })
+            .collect(Collectors.toSet());
+
+        return filteredClaims;
```
[the remark this answers is not in the document]
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java]: I always prefer a return variable for to make debugging easier as you can get the result of a method and have the context it was generated in.

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/DocAssemblyService.java  2
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/MockedDatabaseConfiguration.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/livesupport/BulkPrintNotificationService.java  1
- domain-model/src/test/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtilsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandler.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/LiveSupportEmailTemplates.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimFeatures.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/services/livesupport/BulkPrintNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/errors/EndpointErrorsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/LiveSupportEmailProperties.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/GenerateOrderCallbackHandler.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/MockedDatabaseConfiguration.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimFeatures.java  1
- domain-model/src/test/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtilsTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/DocAssemblyService.java  2
- src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/LiveSupportEmailTemplates.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandler.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/LiveSupportEmailProperties.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/errors/EndpointErrorsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  7
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/services/livesupport/BulkPrintNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/livesupport/BulkPrintNotificationService.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/GenerateOrderCallbackHandler.java  -6
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/NotifyClaimantDefaultCCJJob.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandler.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/livesupport/BulkPrintNotificationService.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/NotifyClaimantDefaultCCJJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/LiveSupportEmailTemplates.java  1
- src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/MockedDatabaseConfiguration.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionsQuestionnaireService.java  6
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/DocAssemblyService.java  2
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/DirectionsQuestionnaireServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/errors/EndpointErrorsTest.java  1
- domain-model/src/test/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtilsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CommonErrors.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimFeatures.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java  5
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandler.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtCSVHeader.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/emails/LiveSupportEmailProperties.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourt.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandlerTest.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/services/livesupport/BulkPrintNotificationServiceTest.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ClaimantResponseContentProvider.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/GenerateOrderCallbackHandler.java  -6
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/NotifyClaimantDefaultCCJJob.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/DocAssemblyService.java  2
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/NotifyClaimantDefaultCCJJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  8
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java  5
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CommonErrors.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/errors/EndpointErrorsTest.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimFeatures.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtCSVHeader.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
- domain-model/src/test/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtilsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  1
- src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/MockedDatabaseConfiguration.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionsQuestionnaireService.java  6
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/DirectionsQuestionnaireServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandler.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourt.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandler.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ClaimantResponseContentProvider.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/GenerateOrderCallbackHandler.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  3
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/support/ResendStaffNotificationsCoreCaseDataTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilter.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/orders/BulkPrintSupportController.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtCSVHeader.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/processors/NoCacheFilterTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandler.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/DocAssemblyService.java  2
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/NotifyClaimantDefaultCCJJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CommonErrors.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionsQuestionnaireService.java  6
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/DrawJudgeOrderCallbackHandlerTest.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimFeatures.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- domain-model/src/test/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtilsTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/DirectionsQuestionnaireServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
- src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/deprectaed/MockedDatabaseConfiguration.java  1
- src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/errors/EndpointErrorsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandler.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  8
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourt.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/support/orders/BulkPrintSupportControllerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/NotifyClaimantDefaultCCJJob.java  3
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/GenerateOrderCallbackHandler.java  -6
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ClaimantResponseContentProvider.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/AbstractStateTransitionJob.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculatorTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculator.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/ScheduledStateTransitionContentProviderTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJob.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ScheduledStateTransitionContentProvider.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJob.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJobTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJobTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJob.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculatorTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculator.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/AbstractStateTransitionJob.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ScheduledStateTransitionContentProvider.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/ScheduledStateTransitionContentProviderTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJob.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculator.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ScheduledStateTransitionContentProvider.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculatorTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJob.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/ScheduledStateTransitionContentProviderTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/AbstractStateTransitionJob.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJob.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJobTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ScheduledStateTransitionContentProvider.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculator.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJob.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/WaitingTransferJob.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/AbstractStateTransitionJob.java  3
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/jobs/cron/StayClaimJobTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionServiceTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/ScheduledStateTransitionContentProviderTest.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitionsTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/statetransition/StateTransitions.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/StateTransitionCalculatorTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4777384&bid=15917552)

**Pull request merged**